core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -506,6 +517,104 @@ private Builder visitAggregate(Aggregate e, List<Integer> groupKeyList,
     return buildAggregate(e, builder, selectList, groupByList);
   }
 
+  /**
+   *  Builds SqlPivotNode for Aggregate RelNode.
+   *
+   * @param e The aggregate node with pivot relTrait flag
+   * @param builder The SQL builder
+   * @param selectColumnList  selectNodeList from Project node
+   * @return  Result with sqlPivotNode wrap in it.
+   */
+  //create method for SqlPivot
+  public Result buildSqlPivotNode(Aggregate e, Builder builder, List<SqlNode> selectColumnList) {
```

Comment:
can we have something similar to  **UnpivotRelToSqlUtil** ,and move all the supporting methods like **getAxisSqlNodes** to that util class

Reply:
Done